File: Leetcode/1233 - Remove Sub-Folders from the Filesystem/solution.cpp
vector<string> removeSubfolders(vector<string> &folders)
{
    sort(folders.begin(), folders.end());
    vector<string> result;
    for (const string &folder : folders)
    {
        if (result.empty() || folder.find(result.back() + '/') != 0)
        {
            result.push_back(folder);
        }
    }
    return result;
}

File: Leetcode/1233 - Remove Sub-Folders from the Filesystem/solution.ts
function removeSubfolders(folderss: string[]): string[] {
    folders.sort();
    const result: string[] = [];

    for (const folder of folders) {
        if (result.length === 0 || !folder.startsWith(result[result.length - 1] + '/')) {
            result.push(folder);
        }
    }

    return result;
};